### DISCUSSION TITLE
Access OC database from custom program?

### DISCUSSION THREAD

**Discussion by MikeKry** [OrchardCMS/OrchardCore#9621] [Q&A]
Is it possible to create custom program (e.g. console app) which would use OC services? 

I am thinking of creating microservice with access to DB and I have been trying to use OC libraries to setup DB connection. So far I have ended with using builder.AddDataAccess and manually injecting other services, but I would like to know if there is some better way because right now it looks impossible to inject ShellSettings. 

Perhaps there is some documentation how to do that? 

My goal is to run example from this page: https://docs.orchardcore.net/en/dev/docs/reference/core/Data/
but in console app with use of OC nugets.

**Comment by hishamco** [comment]
OC Database based on YesSQL which can accessible to run on Web / Windows or Console app, perhaps the Data layer in OC have some services to inject. Could you please ellaborate what you are trying to do or why you need to access the database from Console App?

BWT almost the services are exosed as Graph QL which make it easier to consume from any app

**Reply by MikeKry** [reply]
@hishamco 

Yes, I know that I could access data via graphql or restapi, but in this case, I wanted to create some console app that would be in same folder as project, it would load same tenant settings as OC and perform DB operation based on that settings. 

If it is not ready for that use case, then it does not matter, I will do it some other way.

**Reply by deanmarcussen** [reply]
Orchard Code is mostly predicated on being a web framework application, so things like the curent shell, are largely driven by middleware. Which you wouldn't haven any of in a console app.

That being said, I've quite successfully built apis, that connect to a YesSQL database, and have nothing to do with OC, yet are still driven by another site running as an OC admin/editor. 

Just connect to YesSQL directly, or go down the route of registering your own shell settings accessor (`IShellSettingsSources` etc)
